### PR TITLE
[Component with its own class name as service type] LPS-194841 Refactor Interpreter component 

### PR DESCRIPTION
**Pull request by Valmir-unicap**

### CODE REVIEW

**Comment by Valmir-unicap** [comment]
ci:test:sf

**Comment by Valmir-unicap** [comment]
ci:test:sf

File: modules/apps/sharing/sharing-web/src/main/java/com/liferay/sharing/web/internal/interpreter/SharingEntryInterpreterProviderImpl.java [hunk 1]
```diff
@@ -71,8 +72,11 @@ private boolean _isAssetObject(String className) {
 	}
 
 	@Reference
-	private AssetRendererSharingEntryInterpreter
-		_assetRendererSharingEntryInterpreter;
+	private AssetEntryLocalService _assetEntryLocalService;
+
+	private final AssetRendererSharingEntryInterpreter
```
Comment (MrJohn1911) [inline modules/apps/sharing/sharing-web/src/main/java/com/liferay/sharing/web/internal/interpreter/SharingEntryInterpreterProviderImpl.java:77]: Please initialize this field in the activate method.

File: modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/handler/AMImageRequestHandler.java [hunk 2]
```diff
@@ -286,6 +287,9 @@ private void _processAMImage(
 	private AMImageFinder _amImageFinder;
 
 	@Reference
-	private PathInterpreter _pathInterpreter;
+	private DLAppService _dlAppService;
+
+	private final PathInterpreter _pathInterpreter = new PathInterpreter(
```
Comment (MrJohn1911) [inline modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/handler/AMImageRequestHandler.java:292]: Please initialize this field in the activate method.

File: modules/apps/adaptive-media/adaptive-media-image-impl/src/test/java/com/liferay/adaptive/media/image/internal/handler/PathInterpreterTest.java [hunk 3]
```diff
@@ -232,6 +232,7 @@ public void testNullPathInfoFails() {
 	private final DLAppService _dlAppService = Mockito.mock(DLAppService.class);
 	private final FileEntry _fileEntry = Mockito.mock(FileEntry.class);
 	private final FileVersion _fileVersion = Mockito.mock(FileVersion.class);
-	private final PathInterpreter _pathInterpreter = new PathInterpreter();
+	private final PathInterpreter _pathInterpreter = new PathInterpreter(
+		_amImageConfigurationHelper, _dlAppService);
```
Comment (MrJohn1911) [inline modules/apps/adaptive-media/adaptive-media-image-impl/src/test/java/com/liferay/adaptive/media/image/internal/handler/PathInterpreterTest.java:236]: Please initialize this field in the activate method.

**Review by MrJohn1911 — CHANGES_REQUESTED** [review]

**Comment by Valmir-unicap** [comment]
ci:test:sf

**Comment by Valmir-unicap** [comment]
ci:test:sf

File: modules/apps/adaptive-media/adaptive-media-image-impl/src/test/java/com/liferay/adaptive/media/image/internal/handler/PathInterpreterTest.java [hunk 4]
```diff
@@ -40,6 +40,9 @@ public class PathInterpreterTest {
 
 	@Before
 	public void setUp() {
+		_pathInterpreter = new PathInterpreter(
+			_amImageConfigurationHelper, _dlAppService);
+
 		ReflectionTestUtil.setFieldValue(
```
Comment (rafaprax) [inline modules/apps/adaptive-media/adaptive-media-image-impl/src/test/java/com/liferay/adaptive/media/image/internal/handler/PathInterpreterTest.java:46]: remove the reflection usage here

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Review by rafaprax — APPROVED** [review]

**Comment by Valmir-unicap** [comment]
sent to the team repository

**Pull request closed without merging**